engine/table/src/main/java/io/deephaven/engine/table/impl/BlinkTableTools.java

```diff
@@ -41,11 +38,31 @@ public class BlinkTableTools {
     public static Table blinkToAppendOnly(final Table blinkTable) {
         final UpdateGraph updateGraph = blinkTable.getUpdateGraph();
         try (final SafeCloseable ignored = ExecutionContext.getContext().withUpdateGraph(updateGraph).open()) {
-            return internalBlinkToAppendOnly(blinkTable);
+            // Setting the row limit as maximum allowed value
+            return internalBlinkToAppendOnly(blinkTable, Long.MAX_VALUE);
         }
     }
 
-    private static Table internalBlinkToAppendOnly(final Table blinkTable) {
+    /**
+     * Convert a Blink Table to an in-memory append only table with a limit on maximum row count. Any updates beyond
+     * that limit won't be appended to the table.
+     *
+     * @param blinkTable The input blink table
+     * @param rowLimit The maximum number of rows in the append-only table
+     * @return An append-only in-memory table representing all data encountered in the blink table across all cycles
+     *         till maximum row count
+     */
+    public static Table blinkToAppendOnly(final Table blinkTable, long rowLimit) {
```

Comment:
You might consider renaming `rowLimit` to `sizeLimit` or similar; we want to be very clear that this isn't about the maximum row key to copy or anything like that.